pallets/eth-app/src/lib.rs

```diff
@@ -86,12 +91,32 @@ decl_module! {
             let who = ensure_signed(origin)?;
             let asset_id: BridgedAssetId = H160::zero();
 
-            // <asset::Module<T>>::do_burn(asset_id, &who, amount)?;
+            let transfer_event = RawEvent::Transfer(who.clone(), recipient, amount);
             let asset_id = <asset::Module<T>>::get_native_asset_id(asset_id);
-            let result = <assets::Module<T>>::assets_burn(&asset_id, &who, &amount.low_u128().saturated_into::<T::Balance>());
 
-            ensure!(result.is_ok(), Error::<T>::BurnFailure);
-            Self::deposit_event(RawEvent::Transfer(who, recipient, amount));
+            let amount = amount.low_u128().saturated_into::<BalanceOf<T>>();
+
+            ensure!(T::Currency::can_slash(
+                asset_id,
+                &who,
+                amount
+                ),
+                Error::<T>::BurnFailure);
+
+            let new_balance = T::Currency::free_balance(asset_id, &who) - amount;
+            T::Currency::ensure_can_withdraw(asset_id,
+                &who,
+                amount,
+                WithdrawReasons::all(),
+                new_balance).or(Err(Error::<T>::BurnFailure))?;
+
+            T::Currency::slash(
```

Comment:
introduct burn&settle function instead

Reply:
should not use burn directly as its not working properly when one wants to burn more tokens that available

Reply:
good reference: `MultiCurrency::withdraw`